src/lib.rs

```diff
@@ -292,6 +292,26 @@ macro_rules! write_reg {
         use $periph::{*};
         (*$instance).$reg.write($value);
     }};
+    //
+    // Register array
+    //
+    ( $periph:path, $instance:expr, $reg:ident [ $offset:expr ], $( $field:ident : $value:expr ),+ ) => {{
+        #[allow(unused_imports)]
+        use $periph::{*};
+        #[allow(unused_imports)]
+        (*$instance).$reg[$offset].write(
+            $({
+                use $periph::{$reg::$field::{W::*, RW::*}};
+                ($value << { use $periph::{$reg::$field::offset}; offset })
+                    & { use $periph::{$reg::$field::mask}; mask }
+            }) | *
+        );
+    }};
+    ( $periph:path, $instance:expr, $reg:ident [ $offset:expr ], $value:expr ) => {{
+        #[allow(unused_imports)]
+        use $periph::{*};
+        (*$instance).$reg[$offset].write($value);
+    }};
```

Comment:
Each macro pattern for register arrays is a near duplicate of the pattern for an individual register. We might be able to consolidate with two layers of macros:

1. Extract an individual register and calls a second macro.
2. Produce the read and / or write on that individual register.

I haven't explored this. But if we want register arrays without this code duplication, I can give it a shot.